### PR TITLE
[Feat] Stableswap: allow pool admin changes

### PR DESCRIPTION
**Pull request by alienHunterOnMars**
Added new version of stableswap pool which can have a different owner

### CODE REVIEW

**Review by epanchee — CHANGES_REQUESTED** [review]
I think we do not need to fork current stableswap implementation to add one feature. Just add this new feature in the current implementation.

**Comment by alienHunterOnMars** [comment]
> > I think we do not need to fork current stableswap implementation to add one feature. Just add this new feature in the current implementation.
> 
> Agree, but the spec (ASTRO-1771) mentioned that we need a version of the pool with this feature, not essentially want to add this to the generic stableswap pool implementation. Will cross-check what's really needed and proceed accordingly

looks like we don't want to add it to core stableswap code, so then this PR should be okay

File: contracts/pair_stable_owner/src/contract.rs [hunk 1]
```diff
@@ -0,0 +1,1587 @@
...
+pub fn update_config(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    params: Binary,
+) -> Result<Response, ContractError> {
+    let config = CONFIG.load(deps.storage)?;
+    let factory_config = query_factory_config(&deps.querier, config.factory_addr.clone())?;
+
+    if info.sender != factory_config.owner {
```
Comment (epanchee) [inline contracts/pair_stable_owner/src/contract.rs:1469]: Here only the contract owner should be able to change parameters

File: contracts/pair_stable_owner/src/contract.rs [hunk 2]
```diff
@@ -0,0 +1,1587 @@
...
+/// ## Description
+/// Used for contract migration. Returns a default object of type [`Response`].
+/// ## Params
+/// * **_deps** is an object of type [`DepsMut`].
+///
+/// * **_env** is an object of type [`Env`].
+///
+/// * **_msg** is an object of type [`MigrateMsg`].
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn migrate(deps: DepsMut, env: Env, _msg: MigrateMsg) -> StdResult<Response> {
```
[the remark this answers is not in the document]
Comment (ShadoySV) [reply contracts/pair_stable_owner/src/contract.rs:1425]: No, the contract's owner should be set to the factory owner by querying the factory only for this contract version

File: contracts/pair_stable_owner/src/contract.rs [hunk 3]
```diff
@@ -0,0 +1,1587 @@
...
+            label: String::from("Astroport LP token"),
+        }
+        .into(),
+        id: INSTANTIATE_TOKEN_REPLY_ID,
+        gas_limit: None,
+        reply_on: ReplyOn::Success,
+    }];
+
+    Ok(Response::new().add_submessages(sub_msg))
+}
```
[the remark this answers is not in the document]
Comment (ShadoySV) [reply contracts/pair_stable_owner/src/contract.rs:130]: No, should be passed and set by init_params as you suggested below

[the remark this answers is not in the document]
Comment (epanchee) [reply contracts/pair_stable_owner/src/contract.rs:130]: Yep, correct

**Review by epanchee — APPROVED** [review]

**Review by ShadoySV — APPROVED** [review]

**Review by maksk7 — APPROVED** [review]

File: contracts/pair_stable_owner/examples/pair_stable_schema.rs [hunk 4]
```diff
@@ -0,0 +1,40 @@
+use std::env::current_dir;
```
Comment (ShadoySV) [inline contracts/pair_stable_owner/examples/pair_stable_schema.rs:1]: Rename this file `pair_stable_owner_schema.rs`

**Review by ShadoySV — APPROVED** [review]

File: contracts/pair_stable_owner/examples/pair_stable_schema.rs (hunk 4 above)
Comment (maksk7) [reply contracts/pair_stable_owner/examples/pair_stable_schema.rs]: Renamed

**Pull request merged**